realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -437,7 +442,16 @@ private void completedAsyncQueriesUpdate(QueryUpdateTask.Result result) {
                 // refresh the Realm to the version provided by the worker thread
                 // (advanceRead to the latest version may cause a version mismatch error) preventing us
                 // from importing correctly the handover table view
-                realm.sharedGroupManager.advanceRead(result.versionID);
+                try {
+                    realm.sharedGroupManager.advanceRead(result.versionID);
+                } catch (BadVersionException e) {
+                    // It is possible for the SharedGroup on the caller thread to advance before handling the callback.
```

Comment:
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  jcjacksonengineering@gmail.com
    Domain joe-jackson.net has exceeded the max emails per hour (103/100 (103%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext3.iad.github.net ([192.30.252.194]:36636 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by rs5.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.86_1)
	(envelope-from <noreply@github.com>)
	id 1afmK0-001wQS-Lu
	for jcjackson@joe-jackson.net; Tue, 15 Mar 2016 06:35:28 -0400
Date: Tue, 15 Mar 2016 03:33:30 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1458038010;
	bh=c5eYn7xBHMHUjIqTxVhVoenhjhIS77XofXbx8ZRTNfM=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=EiZfI1yra5wleqGdqiHLCS7tTGnbv/V27R0RmUzGl/byOOxl89xV2WhOILR3UT5ly
	 CzrRB2nhYkSgtBKSyI6rqkz8aTJKZtMu4AYxHKXvaUgUQpixTtu6PlfD9VomgefpVs
	 McS7ohQ5p55yGyMW/G78EUOzf38jZO9gNqhy+CSQ=
From: Makoto Yamazaki <notifications@github.com>
Reply-To: realm/realm-java <reply@reply.github.com>
To: realm/realm-java <realm-java@noreply.github.com>
Cc: Joe Jackson <jcjackson@joe-jackson.net>
Message-ID: <realm/realm-java/pull/2450/r56142959@github.com>
In-Reply-To: <realm/realm-java/pull/2450@github.com>
References: <realm/realm-java/pull/2450@github.com>
Subject: Re: [realm-java] Fix BadVersionException (#2450)
Mime-Version: 1.0
Content-Type: multipart/alternative;
 boundary="--==_mimepart_56e7e4fac8092_79923f80cc4552b87393e0";
 charset=UTF-8
Content-Transfer-Encoding: 7bit
Precedence: list
X-GitHub-Sender: zaki50
X-GitHub-Recipient: jcj52436999
X-GitHub-Reason: comment
List-ID: realm/realm-java <realm-java.realm.github.com>
List-Archive: https://github.com/realm/realm-java
List-Post: <mailto:reply@reply.github.com>
List-Unsubscribe: <mailto:unsub+000f94f74178681d8b35ee48849a70e9b019f7bc4041b26e92cf0000000112ffa6fa92a170ce0358ac6f@reply.github.com>,
 <https://github.com/notifications/unsubscribe/AA-U916Up70GCBB9EVEAzIHqSYEWaxEkks5ptor6gaJpZM4HwK21>
X-Auto-Response-Suppress: All
X-GitHub-Recipient-Address: jcjackson@joe-jackson.net


----==_mimepart_56e7e4fac8092_79923f80cc4552b87393e0
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

> @@ -437,7 +442,16 @@ private void completedAsyncQueriesUpdate(QueryUpdateTask.Result result) {
>                  // refresh the Realm to the version provided by the worker thread
>                  // (advanceRead to the latest version may cause a version mismatch error) preventing us
>                  // from importing correctly the handover table view
> -                realm.sharedGroupManager.advanceRead(result.versionID);
> +                try {
> +                    realm.sharedGroupManager.advanceRead(result.versionID);
> +                } catch (BadVersionException e) {
> +                    // It is possible for the SharedGroup on the caller thread to advance before handling the callback.

The comment here is not correct. We should update it.

---
You are receiving this because you commented.
Reply to this email directly or view it on GitHub:
https://github.com/realm/realm-java/pull/2450/files#r56142959
----==_mimepart_56e7e4fac8092_79923f80cc4552b87393e0
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<p>In <a href="https://github.com/realm/realm-java/pull/2450#discussion_r56142959">realm/realm-library/src/main/java/io/realm/HandlerController.java</a>:</p>
<pre style='color:#555'>&gt; @@ -437,7 +442,16 @@ private void completedAsyncQueriesUpdate(QueryUpdateTask.Result result) {
&gt;                  // refresh the Realm to the version provided by the worker thread
&gt;                  // (advanceRead to the latest version may cause a version mismatch error) preventing us
&gt;                  // from importing correctly the handover table view
&gt; -                realm.sharedGroupManager.advanceRead(result.versionID);
&gt; +                try {
&gt; +                    realm.sharedGroupManager.advanceRead(result.versionID);
&gt; +                } catch (BadVersionException e) {
&gt; +                    // It is possible for the SharedGroup on the caller thread to advance before handling the callback.
</pre>
<p>The comment here is not correct. We should update it.</p>

<p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br>You are receiving this because you commented.<br>Reply to this email directly or view it on GitHub:<br><a href="https://github.com/realm/realm-java/pull/2450/files#r56142959">https://github.com/realm/realm-java/pull/2450/files#r56142959</a><img alt="" height="1" src="https://github.com/notifications/beacon/AA-U92I6uj8zPm3WGfaT_ts2zzdWD6ZWks5ptor6gaJpZM4HwK21.gif" width="1" /></p>
<div itemscope itemtype="http://schema.org/EmailMessage">
<div itemprop="action" itemscope itemtype="http://schema.org/ViewAction">
  <link itemprop="url" href="https://github.com/realm/realm-java/pull/2450/files#r56142959"></link>
  <meta itemprop="name" content="View Pull Request"></meta>
</div>
<meta itemprop="description" content="View this Pull Request on GitHub"></meta>
</div>

----==_mimepart_56e7e4fac8092_79923f80cc4552b87393e0--